app/uk/gov/hmrc/gform/validation/ValidationService.scala

```diff
@@ -442,22 +444,23 @@ class ComponentsValidator(
     val WholeShape = "([+-]?)(\\d+(,\\d{3})*?)[.]?".r
     val FractionalShape = "([+-]?)(\\d*(,\\d{3})*?)[.](\\d+)".r
     (TextConstraint.filterNumberValue(value), maxFractional, mustBePositive) match {
-      case (WholeShape(_, whole, _), _, _) if whole.size > maxWhole =>
+      case (WholeShape(_, whole, _), _, _) if filterCommas(whole).size > maxWhole =>
         getError(fieldValue, s"must be at most $maxWhole digits")
       case (WholeShape("-", _, _), _, true) => getError(fieldValue, "must be a positive number")
       case (WholeShape(_, _, _), _, _)      => ().valid
-      case (FractionalShape(_, whole, _, fractional), 0, _) if whole.size > maxWhole && fractional.size > 0 =>
+      case (FractionalShape(_, whole, _, fractional), 0, _)
+          if filterCommas(whole).size > maxWhole && filterCommas(fractional).size > 0 =>
         getError(fieldValue, s"number must be at most $maxWhole whole digits and no decimal fraction")
       case (FractionalShape(_, whole, _, fractional), _, _)
-          if whole.size > maxWhole && fractional.size > maxFractional =>
+          if filterCommas(whole).size > maxWhole && filterCommas(fractional).size > maxFractional =>
```

Comment:
Surely you don't need to filterCommas from the fractional part? No commas are allowed there by FractionalShape